dartagnan/src/main/java/com/dat3m/dartagnan/smt/FormulaManagerExt.java

```diff
@@ -0,0 +1,95 @@
+package com.dat3m.dartagnan.smt;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.sosy_lab.java_smt.api.*;
+
+import java.util.List;
+import java.util.stream.IntStream;
+
+/*
+    Enriches JavaSMT's FormulaManager with
+        - new theories (TupleFormulaManager)
+        - utility functions
+ */
+public class FormulaManagerExt {
+
+    private final FormulaManager fmgr;
+    private final TupleFormulaManager tmgr;
+
+    public FormulaManagerExt(FormulaManager fmgr) {
+        this.fmgr = fmgr;
+        this.tmgr = new TupleFormulaManager(this);
+    }
+
+    public FormulaManager getUnderlyingFormulaManager() { return fmgr; }
```

Comment:
All other methods (the one following) use the same method name as JavaSMT (i.e. without the "underlying"). I don't see why this one should be a special case.

Reply:
For two reasons. First, it IS a special case, because `FormulaManager` is not just another manager like `BooleanFormulaManager` or `IntegerFormulaManager`. `FormulaManager` is the container of those two.
Secondly, the whole point of `FormulaManagerExt` is to wrap `FormulaManager` in such a way that we should ideally never need to access the underlying `FormulaManager` at all. Therefore, giving its getter method a distinct name is deliberate.